src/dlgRoomExits.cpp

```diff
@@ -2286,13 +2302,14 @@ void dlgRoomExits::slot_checkModified()
                                            ? 1 : 0 ;
                     currentExit.weight = pI->text(2).toInt();
                     currentExit.hasStub = false;
-                    if( foundMap.contains( currentCmd )
-                        && foundMap.value( currentCmd )->destination == currentExit.destination
-                        && foundMap.value( currentCmd )->door        == currentExit.door
-                        && foundMap.value( currentCmd )->hasNoRoute  == currentExit.hasNoRoute
-                        && foundMap.value( currentCmd )->weight      == currentExit.weight      )
+                    auto exit = foundMap.value(currentCmd);
```

Comment:
Depending on the nature of the nesting, but if it both includes ORs as well as ANDs or other parenthesised forms there could still be the same level of indentation for the code and the immediately preceding condition line - fair enough this is a bit contrived but I am not sure we can _consistently_ have a layout pattern where the indentation of the conditions and code do not match:

        if (! exit
            || (exit->destination == currentExit.destination
                && exit->door        == currentExit.door
                && exit->hasNoRoute  == currentExit.hasNoRoute
                && exit->weight      == currentExit.weight      ) {
                foundMap.remove( currentCmd );
        } else {
            isModified = true;
            break;
        }